src/ngraph_nnvm_ops.cc

```diff
@@ -82,31 +67,60 @@ void compile_if_needed(std::shared_ptr<Graph> graph, int mode) {
   }
 }
 
+template <typename T>
+std::vector<bool> is_bool(const T& nodes) {
+  std::vector<bool> bools;
+  for (const auto& node : nodes) {
+    bools.push_back(node->get_element_type() == ngraph::element::boolean);
+  }
+  return bools;
+}
+template <typename T>
+std::vector<bool> is_scal(const T& nodes) {
```

Comment:
is this `is_scalar` ? can we use full function name

Reply:
There's an ngraph::is_scalar I was getting odd naming conflicts with, never went back to debug. Will try again.